### ISSUE TITLE
Bug: Data Files search requires extra action to load current submission

### ISSUE THREAD

**Issue by jtimpe** [raft-tech/TANF-app#3354]
### Description

Clicking **"Search"** on the **Data Files** page reveals the hidden results section, but **results are not populated immediately**. Sometimes, users must either search again or navigate to the **Submission History** tab and back before the **Current Submission** properly loads.

While this was reported in [PR #1577](https://github.com/raft-tech/TANF-app/pull/1577), the issue has been observed locally prior to the dependency upgrades, so it may not be directly related.

#### Screenshot:
![Data Files Issue](https://github.com/user-attachments/assets/b9910662-9ab0-4207-b846-b684e3082e50)

### Action Taken

Navigated to the **Data Files** page, selected filters, and clicked **"Search"**. Observed whether the **Current Submission** and **Submission History** populated as expected.

### What I expected to see

- Both **Current Submission** and **Submission History** populate immediately after clicking **"Search"**, with no need to click away or search again.

### What I did see

- **Current Submission** sometimes did not load.  
- Needed to either click **"Search"** again or navigate away and back (to/from Submission History) for the results to appear.

### Other Helpful Information

+ URL of the page I was on: Data Files page  
+ Browser and version: _Chrome, Firefox (versions TBD)_  
+ Operating System: (Windows | MacOS X | Linux | Other)  
+ Is the issue repeatable?: yes  
+ Has the issue occurred more than once?: yes

### Usability Testing Criteria
<details>
  <summary>Test Cases for UX</summary>

#### Test Case 1: Immediate Loading of Current Submission After Search
- [ ] On the Data Files page, apply valid filter criteria
- [ ] Click **"Search"**

##### Expected Outcomes
- **Current Submission** panel is visible  
- Results load immediately without needing a second click  
- No flickering, delay, or empty results panel

#### Test Case 2: Submission History Also Loads Immediately
- [ ] On the same search, observe the **Submission History** tab

##### Expected Outcomes
- **Submission History** data is visible and correctly populated  
- No need to toggle tabs or refresh the view  
- History matches expectations based on selected filters

#### Test Case 3: No Re-click Required to Trigger Data Load
- [ ] After clicking **"Search"**, do not interact further
- [ ] Observe whether all data loads without repeating the action

##### Expected Outcomes
- No repeated user actions (re-clicking or tab switching) are required  
- Data appears automatically on the first submission attempt  
- Page state reflects correct, active filters

#### Test Case 4: Tab Switching Behavior
- [ ] Toggle between **Current Submission** and **Submission History** tabs

##### Expected Outcomes
- Data is retained and not re-fetched unnecessarily  
- Switching tabs does not “trigger” delayed rendering that should have occurred earlier  
- Tab states update cleanly without visual glitches

#### Test Case 5: Regression Check
- [ ] Repeat all above tests in:
  - Chrome and Firefox (latest versions)
  - Across Mac and Windows (if possible)

##### Expected Outcomes
- Consistent behavior across platforms  
- No regressions in expected rendering or user experience  
- Filters, tabs, and results all function smoothly

</details>

**Comment by jtimpe** [comment]
I believe flux (action/reducer/store pattern used by redux) contributes to this issue. I think refactoring to the context api (as mentioned by the tech memo ticket #3305) would make the data flow more understandable.

**Comment by lhuxraft** [comment]
Could be replaced from tickets that come as an outcome from #3305. Not confident that this behavior is related to the react upgrade

**Comment by elipe17** [comment]
@jtimpe still needed? Did this go away when we removed the download option?

**Comment by jtimpe** [comment]
@elipe17 right, we no longer have a preview for the current file so this is irrelevant
